OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Distributed;
+using OrderService.Domain.Abstractions.Data;
+using OrderService.Domain.Entities;
+using System.Text.Json;
+
+namespace OrderService.Infrastructure.Services;
+
+internal class RedisStorageService : ITemporaryStorageService
+{
+    private readonly IDistributedCache _cache;
+    private readonly HttpContext _httpContext;
+
+    public RedisStorageService(IDistributedCache cache, IHttpContextAccessor httpContextAccessor)
+    {
+        _cache = cache;
+        _httpContext = httpContextAccessor.HttpContext;
+    }
+ 
+    public async Task<Dictionary<int, ProductEntity>> GetCartAsync(CancellationToken cancellationToken = default)
+    {
+        var userId = _httpContext.User.FindFirst("Id")?.Value;
+        var cartId = GetCartId();
+
+        if (userId is not null)
+        {
+            var oldCartId = _httpContext.Request.Cookies["CartId"];
+
+            if (oldCartId is not null)
+            {
+                var oldCart = await _cache.GetStringAsync(oldCartId, cancellationToken);
+
+                _httpContext.Response.Cookies.Delete("CartId");
+
+                if (oldCart is not null)
+                {
+                    await _cache.RemoveAsync(oldCartId, cancellationToken);
+
+                    var options = new DistributedCacheEntryOptions
+                    {
+                        AbsoluteExpirationRelativeToNow = TimeSpan.FromDays(1)
+                    };
+
+                    await _cache.SetStringAsync(userId, oldCart, options, cancellationToken);
+                }
+            }
+        }
```

Comment:
can you explain this logic, please?

Reply:
This code is used to assign a cart to a userId if the user is authorized. I'm checking if there is a cartId in the cookies, and if it exists, I'm deleting it from the cookies and assigning this cart to the userId

Reply:
what do you think about moving it to a separate method?